routing/register.go

```diff
@@ -51,3 +53,10 @@ func isPasswordValid(password string) bool {
 	}
 	return exp.Match([]byte(password))
 }
+
+func userAlreadyExists(username *string) bool {
+	var res models.User
+	usersCollection := db.Client.Database(os.Getenv("DB_NAME")).Collection(os.Getenv("USERS_COLLECTION"))
```

Comment:
I would save `db.Client.Database(os.Getenv("DB_NAME"))` to a variable so you don't have to type this every time (it's unlikely we'll ever use another database). And `os.Getenv("USERS_COLLECTION")` doesn't need to be a .env variable imo